main.cpp

```diff
@@ -51,6 +53,137 @@
  */
 
 #include <iostream>
+
+
+  struct FloatType 
+  {
+       float add(float lhs, float rhs );
+       float subtract(float lhs,float rhs );
+       float multiply(float lhs,float rhs );
+       float divide(float lhs, float rhs );
+  };
+
+   struct DoubleType 
+   {
+        double add(double lhs, double rhs );
+        double subtract(double lhs,double rhs );
+        double multiply(double lhs,double rhs );
+        double divide(double lhs, double rhs );
+  };
+
+   struct IntType 
+   {
+        int add(int lhs,int rhs );
+        int  subtract(int lhs,int rhs );
+        int multiply(int lhs,int rhs );
+        int divide(int lhs,int rhs );
+  };
+
+  float FloatType::add( float lhs, float rhs )
+  {
+      
+     float var = lhs + rhs;
+
+      return  var ;
+  }
+
+  float FloatType::subtract( float lhs, float rhs )
+  {
+    float var = lhs - rhs;
+
+      return var ;
+  }
+
+  float FloatType::multiply( float lhs, float rhs )
+  {
+    float var = lhs * rhs;
+
+      return var ;
+  }
+
+  float FloatType::divide(float lhs, float rhs )
+{
+
+    float var = lhs / rhs;
+
+    if (rhs == 0.f)
+    { 
+        std::cout << "warning, floating point division by zero returns 'inf' !" << std::endl; 
+    }
+
+    return var;
+}
+
+  double DoubleType::add( double lhs, double rhs )
+  {
+    double var = lhs + rhs;
+
+      return var;
+  }
+
+  double DoubleType::subtract( double lhs, double rhs )
+  {
+    double var = lhs - rhs;
+
+      return var;
+  }
+
+  double DoubleType::multiply( double lhs, double rhs )
+  {
+    double var = lhs * rhs;
+
+      return var;
+  }
+
+  double DoubleType::divide( double lhs, double rhs )
+{
+    double var = lhs / rhs;
+
+    return var;
+}
+
+    int IntType::add( int lhs, int rhs )
+  {
+    int var = lhs + rhs;
+
+      return var;
+  }
+
+  int IntType::subtract( int lhs, int rhs )
+  {
+    int var = lhs - rhs;
+
+      return var;
+  }
+
+  int IntType::multiply( int lhs, int rhs )
+  {
+    int var = lhs * rhs;
+
+      return var;
+  }
+
+  int IntType::divide( int lhs, int rhs )
+{
+
+    if (rhs == 0)
+    {
+        std::cout << "error, integer division by zero will crash the program!" << std::endl;
+
+        std::cout << "returning lhs" << std::endl;
+
+        return lhs; 
+    } 
+    else 
+    {
+        int var = lhs / rhs; 
+
+        return var;
+    }
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise

Reply:
Thank you this is really helpful. This saves a lot of execution steps.